controllers/assignments.py

```diff
@@ -350,33 +403,60 @@ def student_autograde():
     """
     assignment_id = request.vars.assignment_id
     timezoneoffset = session.timezoneoffset if 'timezoneoffset' in session else None
-    assignment = db(db.assignments.id == assignment_id).select().first()
-    if assignment:
-        count = do_autograde(assignment, auth.user.course_id, auth.user.course_name,
-            auth.user.username, None, 'false', timezoneoffset, db, settings)
-        return json.dumps({'message': "autograded {} items".format(count)})
+
+
+    res = _autograde(student_rownum=auth.user.id,
+                     assignment_id=assignment_id,
+                     timezoneoffset=timezoneoffset)
+
+    if not res['success']:
+        session.flash = "Failed to autograde questions for user id {} for assignment {}".format(auth.user.id, assignment_id)
+        res = {'success':False}
     else:
-        return json.dumps({'success': False, 'message': "Could not find this assignment -- This should not happen"})
+        if settings.coursera_mode:
+            res2 = _calculate_totals(student_rownum=auth.user.id, assignment_id=assignment_id)
+            if not res2['success']:
+                session.flash = "Failed to compute totals for user id {} for assignment {}".format(auth.user.id, assignment_id)
+                res = {'success':False}
+            else:
+                _try_to_send_lti_grade(auth.user.id, assignment_id)
+    return json.dumps(res)
+    # assignment = db(db.assignments.id == assignment_id).select().first()
```

Comment:
Is this experimental?  If we don't want this then lets delete the code.

Reply:
`_try_to_send_lti_grade` is not experimental. It checks to see whether lti is set up for this server and for this student/assignment pair. If so, it sends the grade.

Reply:
OK, then there is no reason to keep the commented out code is there?

Reply:
I think the commented out code was removed in the other pull request...